src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -755,7 +755,7 @@ internal void AddResponseFileCommandsForSwitchesSinceInitialReleaseThatAreNeeded
             commandLine.AppendPlusOrMinusSwitch("/publicsign", _store, nameof(PublicSign));
             commandLine.AppendSwitchIfNotNull("/runtimemetadataversion:", RuntimeMetadataVersion);
             commandLine.AppendSwitchIfNotNull("/checksumalgorithm:", ChecksumAlgorithm);
-            commandLine.AppendSwitchIfNotNull("/instrument:", InstrumentationKinds);
+            commandLine.AppendSwitchWithSplitting("/instrument:", Instrument, ",", ';', ',');
```

Comment:
AppendSwitchWithSplitting divides up a list of options separated by commas or semicolons--it's what is done for NoWarn. Simply adding the property value results in it being quoted when round-tripped through GenerateResponseFileContents.